### DISCUSSION TITLE
Best way to show an array of images?

### DISCUSSION THREAD

**Discussion by zasuh** [vercel/next.js#20458] [Help]
Hi!
I'm making a photography site with NextJS. Each page will display about 10-12 images in a grid. What is the best way to load these images onto each page? Do I just put the images into `public/static` and load them from there? Or does Vercel offer some solution where I can query the images and get them from somewhere else?

**Comment by TheDarkStrix** [comment]
Hi , 

I would consider doing all this dynamically, eg: Images are being fetched from a API.
Paginated data coming in from the api, make a request for the image only when the user wants it , this can be achieved using Interceptor Observers and Lazy loading for images that are being loaded. 

Another thing you try is using GetServerSideProps [Next JS Feature] and GetStaticPath to load the data quicker without recompiling project on every addition of a new image, you can read more on this [here](https://nextjs.org/docs/basic-features/data-fetching) .

Hope this helped !

**Reply by zasuh** [reply]
Where would be the best place to store images and pull them using an API? What does the community recommend?

**Reply by TheDarkStrix** [reply]
I would recommend Amazon S3 or Cloudinary or other such services
